### PR TITLE
[Merged by Bors] - feat: port NumberTheory.Cyclotomic.Gal

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by riccardobrasca** [comment]
Having a look.

**Comment by riccardobrasca** [comment]
It should now work. I don't exactly understand why but `generalize_proofs` behave differently.

File: Mathlib/NumberTheory/Cyclotomic/Gal.lean [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+/-- Maps `μ` to the `alg_equiv` that sends `is_cyclotomic_extension.zeta` to `μ`. -/
+noncomputable def fromZetaAut : L ≃ₐ[K] L :=
+  let hζ := (zeta_spec n K L).eq_pow_of_pow_eq_one hμ.pow_eq_one n.pos
+  (autEquivPow L h).symm <|
+    ZMod.unitOfCoprime hζ.choose <|
+      ((zeta_spec n K L).pow_iff_coprime n.pos hζ.choose).mp <| hζ.choose_spec.2.symm ▸ hμ
+#align is_cyclotomic_extension.from_zeta_aut IsCyclotomicExtension.fromZetaAut
+
+set_option pp.proofs true in
```
Comment (Ruben-VandeVelde) [inline Mathlib/NumberTheory/Cyclotomic/Gal.lean:159]:
```suggestion
```

**Comment by riccardobrasca** [comment]
I don't have time anymore, I didn't fix docs/names.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**